### ISSUE TITLE
Extend AngleSharp to allow parsing of custom tags

### ISSUE THREAD

**Issue by SOHODeveloper** [AngleSharp/AngleSharp#1200]
### Description

I am trying to parse CFML.  The problem is that CFML has mostly void tags and the syntax does not require the /> at the end of the tag start.

I would not mind doing the work to allow for some sort of event/callback when an unknown tag is encountered that would allow for the parse to be modified to indicate the tag is a void tag and a few other attributes.

Is there perhaps such a place already in AngleSharp?  If so, please point me to it and how to get started creating such a hook.

### Background

Many companies are wanting to get away from ColdFusion/CFML and this type of feature would allow them to interrogate their code and do some automated conversion.

### Specification

None

**Comment by FlorianRappl** [comment]
No this is not in the spec - I think what you want is a custom syntax.

(Of course we could talk about an extension / flag that could be used, but it needs to be without any performance implications. Also this feature would need to be sponsored.)

**Comment by SOHODeveloper** [comment]
Can you just point me in the right direction.  I would like to extend AngleSharp in a way that I could make it modular and easily update it as you update the core.

I found the Create method in HtmlElementFactory and see where the HtmlUnknownElement is created.   Could I safely plugin a call back (delegate, event) at that point that could make the correct decisions as to the element type and return the creator for it?

Or is there a better place to do this work?   If that is a good place, it would only be performance impacting when requested and when a non-standard tag is encountered.

Bottom line I need to get out of this is recoginizing void elements that don't end in />.   IE  <cfset var=expr> instead of <cfset var=expr/>

**Comment by FlorianRappl** [comment]
No the behavior you are seeking to is deep in the parser. The decision what elements are void / can be self-closed is a fully wired one - as its fully described by the HTML5 spec.

As mentioned we can think about a proper extension to that model, but anything in that direction needs to be sponsored. If the companies you are working with take that migration seriously then they should have no problem of investing in development therein.

Example: https://github.com/AngleSharp/AngleSharp/blob/devel/src/AngleSharp/Html/Parser/HtmlDomBuilder.cs#L1192

**Comment by SOHODeveloper** [comment]
I am assuming that you are saying that me doing the work does not fit into the category of sponsored.

**Comment by FlorianRappl** [comment]
> I am assuming that you are saying that me doing the work does not fit into the category of sponsored.

Not necessarily.

But likewise the work of validating any PR, putting thoughts on performance implications, following up etc. - also needs to be accounted for.

**Comment by SOHODeveloper** [comment]
Thanks for your time.
